src/gui/vsWebSocket.cpp

```diff
@@ -49,20 +49,31 @@ VsWebSocket::VsWebSocket(const QUrl& url, QString token, QString apiPrefix,
     , m_apiPrefix(apiPrefix)
     , m_apiSecret(apiSecret)
 {
-    connect(&m_webSocket, &QWebSocket::connected, this, &VsWebSocket::onConnected);
-    connect(&m_webSocket, &QWebSocket::disconnected, this, &VsWebSocket::onClosed);
-    connect(&m_webSocket, &QWebSocket::disconnected, this, &VsWebSocket::disconnected);
-    connect(&m_webSocket, QOverload<const QList<QSslError>&>::of(&QWebSocket::sslErrors),
-            this, &VsWebSocket::onSslErrors);
-    connect(&m_webSocket, QOverload<QAbstractSocket::SocketError>::of(&QWebSocket::error),
-            this, &VsWebSocket::onError);
-    connect(&m_webSocket, &QWebSocket::textMessageReceived, this,
+    m_webSocket.reset(new QWebSocket());
+    connect(m_webSocket.get(), &QWebSocket::disconnected, this,
+            &VsWebSocket::disconnected);
+    connect(m_webSocket.get(),
+            QOverload<const QList<QSslError>&>::of(&QWebSocket::sslErrors), this,
+            &VsWebSocket::onSslErrors);
+    connect(m_webSocket.get(),
+            QOverload<QAbstractSocket::SocketError>::of(&QWebSocket::error), this,
+            &VsWebSocket::onError);
+    connect(m_webSocket.get(), &QWebSocket::textMessageReceived, this,
             &VsWebSocket::textMessageReceived);
 }
 
+VsWebSocket::~VsWebSocket()
+{
+    if (isValid()) {
+        closeSocket();
+    }
+    m_webSocket->disconnect();
```

Comment:
I don't think it could be null, but probably good to have a `isNull` check around these too